oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/index/ElasticDocument.java

```diff
@@ -147,13 +155,18 @@ public String build() {
                 if (!similarityTags.isEmpty()) {
                     builder.field(ElasticIndexDefinition.SIMILARITY_TAGS, similarityTags);
                 }
+                for (Map.Entry<String, Set<Object>> prop : toStoreInSource.entrySet()) {
+                    if (!properties.containsKey(prop.getKey())) {
+                        builder.field(prop.getKey(), prop.getValue());
+                    }
+                }
```

Comment:
I guess this was made to support regex properties. Although it would work, we should refrain to do that because it will go against the strict mapping we need to soon support.

This is not just a problem related to the excerpt feature but a more general one (eg: regex properties, function properties).

My proposal is to simplify this PR by adding excerpt support for specific properties only (we need to document the difference) and have a separate PR to cover the dynamic properties and aim for a NON-dynamic mapping.

Reply:
Aligned, regex are excluded and a next PR is setting the fields dynamicly created to be only stored. if they are useInExcerpt only.